src/Adapter/Module/CommandHandler/BulkUninstallModuleHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Module\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\Module\Command\BulkUninstallModuleCommand;
+use PrestaShop\PrestaShop\Core\Domain\Module\CommandHandler\BulkUninstallModuleHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Module\Exception\CannotUninstalledModuleException;
+use PrestaShop\PrestaShop\Core\Module\ModuleManager;
+
+#[AsCommandHandler]
+class BulkUninstallModuleHandler implements BulkUninstallModuleHandlerInterface
+{
+    public function __construct(
+        protected ModuleManager $moduleManager,
+    ) {
+    }
+
+    public function handle(BulkUninstallModuleCommand $command): void
+    {
```

Comment:
To keep a uniform behaviour with the other commands we need to assert the presence of the module first, this should be done via the new method introduced in this PR https://github.com/PrestaShop/PrestaShop/pull/37169 `ModuleRepository::getPresentModule`

Also we should start by validating that all provided modules are present and installed, so a check loop should be added at the beginning here:

```suggestion
    {
            foreach ($command->getModules() as $moduleName) {
                // This one will automatically trigger a not found exception if one of the modules is not present
                $module = $this->moduleRepository->getModule($moduleName);
                if (!$module->isInstalled()) {
                    throw new CannotUninstallModuleException(sprintf('Cannot uninstall module %s because it is not installed', $moduleName));
                }
            }
```

I know the module manager already handles similar checks, but the purpose here is to throw the appropriate exception related to the module domain, this way we can catch precise exceptions in the BO when this particular exception is thrown. And on the API side it also allows us to configure some binding between the thrown exception and the appropriate HTTP code.